test/integration/global_deploy_test.rb

```diff
@@ -116,4 +126,54 @@ def test_global_deploy_task_failure
       "Template validation failed",
     ])
   end
+
+  def test_global_deploy_prune_black_box_success
+    assert_deploy_success(deploy_global_fixtures('globals', clean_up: false, selector: 'test=prune1'))
+    reset_logger
+    assert_deploy_success(deploy_global_fixtures('globals', subset: 'storage_classes.yml', selector: 'test=prune1'))
+    assert_logs_match_all([
+      "Phase 1: Initializing deploy",
+      "Using resource selector test=prune1",
+      "All required parameters and files are present",
+      "Discovering resources:",
+      "  - StorageClass/testing-storage-class",
+      "Phase 2: Checking initial resource statuses",
+      %r{StorageClass\/testing-storage-class[\w-]+\s+Exists},
+      "Phase 3: Deploying all resources",
+      %r{Deploying StorageClass\/testing-storage-class[\w-]+ \(timeout: 300s\)},
+      "Don't know how to monitor resources of type StorageClass.",
```

Comment:
I'd remove these next two lines--the fact that we don't yet support this class properly isn't relevant to this test